helix-core/src/syntax.rs

```diff
@@ -752,65 +803,54 @@ pub struct Loader {
 }
 
 impl Loader {
-    pub fn new(config: Configuration) -> Self {
-        let mut loader = Self {
-            language_configs: Vec::new(),
-            language_server_configs: config.language_server,
-            language_config_ids_by_extension: HashMap::new(),
-            language_config_ids_by_suffix: HashMap::new(),
-            language_config_ids_by_shebang: HashMap::new(),
-            scopes: ArcSwap::from_pointee(Vec::new()),
-        };
+    pub fn new(config: Configuration) -> Result<Self, globset::Error> {
```

Comment:
Switching this to return a Result makes the error handling for config (re)loading a little bit messier. I think it's revealing an odd separation between `Configuration` and `Loader` that makes error handling awkward, and it might be a nice chance for a refactor.

I pushed up some ideas in this commit: https://github.com/the-mikedavis/helix/commit/6c455ec4c4b250c8170d632c6a3436b51c0e8f85

The main thing I want to avoid is handling the `Loader::new` error in `Application::new`. If either the Configuration and Loader creation fails we should report on it in this part of `main`: https://github.com/helix-editor/helix/blob/bbcc89241fd63b460ea7b1661a550f5b9eacba4c/helix-term/src/main.rs#L148-L155

To do that we basically handle `Loader` everywhere other than `helix_core` (or the health-check part of `xtask`) rather than `Configuration`, which I think is cleaner anyways.